http4s/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/Config.scala

```diff
@@ -0,0 +1,293 @@
+package com.snowplowanalytics.snowplow.collectors.scalastream
+
+import java.nio.file.{Files, Path}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.collection.JavaConverters._
+
+import com.typesafe.config.ConfigFactory
+
+import io.circe.config.syntax._
+import io.circe.generic.semiauto._
+import io.circe.Decoder
+import io.circe._
+
+import cats.implicits._
+
+import cats.effect.Sync
+
+import com.monovore.decline.{Command, Opts}
+
+object Config {
+
+  def parse[F[_]: Sync](
+    args: List[String]
+  ): F[Config] = {
+    val appConfig = Opts.option[Path]("config", "Path to HOCON configuration", "c", "config.hocon")
+    val parser    = Command[Path]("collector", "Snowplow application that collects tracking events")(appConfig)
+    for {
+      path <- parser.parse(args) match {
+        case Left(help) =>
+          Sync[F].raiseError(new IllegalArgumentException(s"can't read CLI arguments. $help"))
```

Comment:
This is OK for production environments... but I don't think it's the best user experience to get an exception and a stack trace if the command line is invalid.  For the lake loader I used `CommandIOApp` instead of `IOApp` and I was pleased with how it worked out.  It's documented [here](https://ben.kirw.in/decline/effect.html).  If the user supplies the wrong command line config then it handles showing a helpful error message and a non-zero exit code.

Reply:
I was only focusing on knowing the error to solve the issue, not on the style. I'll make it look nicer like you did.